app/models/manager_refresh/inventory_object_lazy.rb

```diff
@@ -36,7 +36,7 @@ def transitive_dependency?
       # and a :stack is a relation to another object, in the InventoryObject object,
       # then this relation is considered transitive.
       !!(key && (inventory_collection.dependency_attributes.keys.include?(key) ||
```

Comment:
- `keys.include?(key)` seems like an anti-pattern, unless depedency_attributes is not a Hash?  I would expect the .include? directly on the dependency_attributes.
- Additionally, you don't need the `!!` at all, and I'm wondering if all that extra `!!()` symbols are contributing to this being confusing.
- If nil can never an actual "key", then there's no need to check for it initially either as both the Hash check and the reflect_on_association can handle nils.

Given all that, perhaps this is easier to read:

```ruby
def transitive_dependency?
  inventory_collection.dependency_attributes.include?(key) ||
    inventory_collection.model_class.try(:reflect_on_association, key)
end